[core] Remove `styled` from `@material-ui/styles` in core (#26101)
mui-org/material-ui: docs/src/pages/guides/migration-v4/migration-v4.md

  +</Stepper>
  ```

> ### styled

Author: Will be moved under `### @material-ui/core/styled` after https://github.com/mui-org/material-ui/pull/26100 is merged

  +import { ServerStyleSheets } from '@material-ui/styles';
  ```

#### styled

> - The `styled` utility has been removed from the `@material-ui/core/styles` package. You can use `@material-ui/styles/styled` instead. Make sure to add a `ThemeProvider` at the root of your application, as the `defaultTheme` is no longer available. If you are using this utility together with `@material-ui/core`, it's recommended you use the `ThemeProvider` component from `@material-ui/core/styles` instead.

Reviewer: ```suggestion
- The `styled` JSS utility is no longer exported from `@material-ui/core/styles`. You can use `@material-ui/styles/styled` instead. Make sure to add a `ThemeProvider` at the root of your application, as the `defaultTheme` is no longer available. If you are using this utility together with `@material-ui/core`, it's recommended you use the `ThemeProvider` component from `@material-ui/core/styles` instead.
```
